src/components/ThoughtAnnotation.tsx

```diff
@@ -279,13 +284,78 @@ const ThoughtAnnotationContainer = React.memo(
       setCalculateContexts(true)
     }, [])
 
+    const positionAnnotation = useCallback(() => {
+      if (editableRef.current && annotationRef.current) {
```

Comment:
You can invert this condition and `return` to short circuit the function and make this a little cleaner by avoiding the extra indentation level.